nginx.conf

```diff
@@ -26,18 +26,37 @@ server {
     location / {
         include uwsgi_params;
         uwsgi_pass unix:///tmp/kolibri_uwsgi.sock;
-        error_page 502 = @502;
+        error_page 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 421 422 423 424 426 428 429 431 451 500 501 502 503 504 505 506 507 508 510 511 @error500;
     }
+    location @error500 {
+        ssi on;
+        internal;
+        root /etc/kolibri/dist/error_pages;
+
+
+        set $error500 /error.html;
+        # Spanish (Latinamerica)
+        if ($request_uri ~ "^\/es-419\/") {
+            set $error500 /la/error.html;
+        }
+        # Spanish (Spain)
+        if ($request_uri ~ "^\/es-es\/") {
+            set $error500 /es-ES/error.html;
+        }
+
+        # Arabic
+        if ($request_uri ~ "^\/ar\/") {
+            set $error500 /ar/error.html;
+        }
+
+        # French
+        if ($request_uri ~ "^\/fr-fr\/") {
+            set $error500 /fr/error.html;
+        }
```

Comment:
I think it might be preferable eventually for some of this code to live in the main Kolibri repo (for maintainability), and have a build step to generate these static HTML files using our translation pipeline for strings.

For similar reasons I think it would also be good to generate these base nginx configurations from Kolibri itself, so that it can be kept up to date with the way that Kolibri generates URLs.

Reply:
The debian installer knows about the location of these files and can put such a path in the nginx.conf.

It's probably best for reusability and code sharing to have such a file maintained in "upstream" kolibri. The main trick is to avoid relying too much on the Kolibri installation, since the error page is supposed to be a fallback when a Kolibri installation doesn't work. So the isolation is desirable.

Reply:
I think we are losing the focus here. This is a simple error page. These comments will bring something simple to an overengineering process that I don't think it's worth for the purpose of showing an error to the user.
Also I think that nginx error pages are and should be out of the scope of the kolibri sources. Some users will use apache, some users in other OS can't use uwsgi....

Reply:
Okay, yes. The original requirements of this project were to have the i18n pipeline independent of Kolibri, so let's move forward with that assumption. (We already have a number of independent projects in Crowdin – windows and soon mac – so this is at least consistent. We can always revisit in the future.)

My main comment originally was simply that for the default case of no path specified, we should inspect the header to get the language.